dio/lib/src/dio_event_processor.dart

```diff
@@ -87,8 +89,76 @@ class DioEventProcessor implements EventProcessor {
 
     return SentryResponse(
       headers: _options.sendDefaultPii ? headers : null,
-      bodySize: dioError.response?.data?.length as int?,
+      bodySize: _getBodySize(
+        dioError.response?.data,
+        dioError.requestOptions.responseType,
+      ),
       statusCode: response?.statusCode,
+      data: _getResponseData(
+        dioError.response?.data,
+        dioError.requestOptions.responseType,
+      ),
     );
   }
+
+  /// Returns the response data, if possible according to the users settings.
+  Object? _getResponseData(Object? data, ResponseType responseType) {
+    if (!_options.sendDefaultPii) {
+      return null;
+    }
+    if (data == null) {
+      return null;
+    }
+    switch (responseType) {
+      case ResponseType.json:
+        final js = json.encode(data);
```

Comment:
Can you check if we can use the `utf8JsonEncoder.convert(data);` from the Dart SDK? the reason is that already has a `jsonSerializationFallback` for stuff that isn't serializable, just a self-defense mechanism.

Reply:
The `utf8JsonEncoder` is not publicly exposed, so we can't use it in the dio package undless we expose/copy the code. Also according to the documentation, it already does call `toString()` if it is not encodable, just like we do with the fallback.

```
If [toEncodable] is omitted, it defaults to a function that returns the
result of calling `.toJson()` on the unencodable object.
 ```

Since `null` is a valid json element, only custom classes will be handled by the fallback and in both cases `.toString()` is called. Am i missing something here?

Reply:
@denrase One of the differences is that the `utf8JsonEncoder` is the `utf8` encoding that  `json.encode` does not do.
`json.encode` defaults to calling `.toJson()` and not `.toString()` per documentation.

Reply:
We can make `utf8JsonEncoder` visible and anotate with `@internal`